phpBB/memberlist.php

```diff
@@ -1147,6 +1147,24 @@
 					$user_rank_data['img'] .= '<br />';
 				}
 			}
+			// include modules for manage groups link display or not
+			// need to ensure the module is active
+			$can_manage_group = false;
+			if ($user->data['is_registered'])
```

Comment:
Check if the user is a group leader here and not after instantiating p_master, listing the modules, and checking if the module is even active. This should get rid of this overhead for most users. Otherwise we'll add this check every time any user accesses this page without any use to it.